### PR TITLE
Support null types for expresion

### PR DESCRIPTION
**Pull request by ngdinhtoan**

### CODE REVIEW

File: expr.go [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 
 import (
+	"database/sql/driver"
 	"fmt"
 	"io"
 	"reflect"
 	"strings"
 )
 
+type nullType interface {
```
Comment (lann) [inline expr.go]: It looks like this is already available as `driver.Valuer`?

File: expr.go [hunk 2]
```diff
@@ -64,6 +69,12 @@ func (eq Eq) toSql(useNotOpr bool) (sql string, args []interface{}, err error) {
 
 	for key, val := range eq {
 		expr := ""
+
+		switch v := val.(type) {
+		case nullType:
+			val, _ = v.Value()
```
Comment (lann) [inline expr.go]: This should pass the error up out of the function, not ignore it.

**Comment by ngdinhtoan** [comment]
@lann, thank for your feedback..

**Comment by lann** [comment]
Looks good, thanks!

**Pull request merged**